tv/tv.cpp

```diff
@@ -131,37 +144,54 @@ struct TVLegacyPass final : public llvm::ModulePass {
       return false;
     }
 
+    if (first || skip_verify) {
+      I->second.fn = move(*fn);
+      if (!opt_always_verify)
+        // Prepare syntactic check
+        I->second.fn_tostr = toString(I->second.fn);
+      printDot(I->second.fn, I->second.n++);
+      return false;
+    }
+
     Transform t;
     t.src = move(I->second.fn);
     t.tgt = move(*fn);
 
-    if (opt_print_dot) {
-      string prefix = to_string(I->second.n++);
-      t.tgt.writeDot(prefix.c_str());
+    std::string src_tostr = move(I->second.fn_tostr);
+    std::string tgt_tostr;
+    if (!opt_always_verify)
+      // Prepare syntactic check
+      tgt_tostr = toString(t.tgt);
+
+    bool regenerate_tgt = verify(t, I->second.n, src_tostr, tgt_tostr);
+
+    if (regenerate_tgt) {
+      I->second.fn = *llvm2alive(F, *TLI);
+      I->second.fn_tostr = toString(I->second.fn);
+    } else {
+      I->second.fn = move(t.tgt);
+      I->second.fn_tostr = move(tgt_tostr);
```

Comment:
this doesn't make sense. when regenerate_tgt is false, we can keep the old string. Because it means that either the two functions are equal or some error occurred (and in that case src isn't valid anymore).

Reply:
Yep, you're right; this isn't necessary.

Then would it be okay if I leave a comment here saying that updating fn_tostr is not necessary due to the reason?
Otherwise, it requires readers to do deep level of interprocedural analysis; to me doing it was painful sometimes.

Reply:
(after removing this move call, I mean)

Reply:
ofc

Reply:
Thx!